src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/CreateSearchParameterBehavior.cs

```diff
@@ -0,0 +1,44 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Fhir.Core.Messages.Create;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search.Parameters
+{
+    public class CreateSearchParameterBehavior<TCreateResourceRequest, TUpsertResourceResponse> : IPipelineBehavior<TCreateResourceRequest, TUpsertResourceResponse>
```

Comment:
Unless the generic is needed for derived types, I think this should still work with:
```csharp
CreateSearchParameterBehavior : IPipelineBehavior<CreateResourceRequest, UpsertResourceResponse>
```

Reply:
I may want to update this for conditionalcreate, so I wanted to leave it generic for now.